### DISCUSSION TITLE
Viewing Old Logs in Kibana

### DISCUSSION THREAD

**Discussion by palitech100** [Security-Onion-Solutions/securityonion#7508] [Unsupported Versions] [answered]
Hello,

I need to review connections related to one internal host that occurred in October of 2021. I am trying to figure out, how I can get that done. I saved the pcaps from that timeframe on an external drive to make sure I don't loose them. I am not sure what the best, most efficient approach would be.

Can I view an old index from that timeframe? If so, how can I do that? Or would you recommend I import the pacps into a new instance of Security Onion? I have a new VM running in import mode, I am using a wildcard to import import several pcaps, but I know this is not an efficient approach. Is it possible to save the pcaps on the new  VM and run tcpreplay? I understand that tcpreplay won't give me the original timestamps, but it may be more efficient?

I think viewing the old index would be the best approach. I am just not sure if that is possible and how to accomplish it. 

I hope you can help me. I look forward to your response.

Thank you.

Martina

**Comment by dougburks** [comment, marked as answer]
If the Elasticsearch index is still on disk but closed, then you can open the index. You would probably have to disable Curator or create an exception so that it doesn't close the index again.

**Reply by palitech100** [reply]
Hi Doug, thank you for your response! Yes the closed indices are still on the disk. Can you please tell me how I disable Curator? And by creating an exception, do you mean editing a yaml file in the /opt/so/conf/curator/action/ and entering the index that I want to keep open? 

Like this:
/opt/so/conf/curator/action/so-elasticsearch-close.yml 

 actions:
      9   1:
     10     action: close
     11     description: >-
     12       Close elasticsearch indices older than 30 days.
     13     options:
     14       delete_aliases: False
     15       timeout_override:
     16       continue_if_exception: False
     17       disable_action: False
     18     filters:
     19     - filtertype: pattern
     20       kind: regex
     21       value: '^(logstash-elasticsearch.*|so-elasticsearch.*)$'
     22     - filtertype: age
     23       source: name
     24       direction: older
     25       timestring: '%Y.%m.%d'
     26       unit: days
     27       unit_count: 30
     28       exclude: **8ozm8173Sh2yMQVTad4nsw**

**Reply by palitech100** [reply]
I am also having problems, when trying to open the closed index, while I am trying to figure out how to disable Curator or create an exception. I get a certificate error and when I run the command without checking the certificate, I get an authentication error. Can you please provide me with guidance? I very much appreciate your help. 

I am running these remotely, so I assume I have to replace localhost with the IP address of the system.

curl -X POST "https://x.x.x.x:9200/ivPDtOJhTwyNzHQe6c2-vQ/_open?pretty"
curl: (60) Peer's Certificate issuer is not recognized.
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.

curl -X POST "https://x.x.x.x:9200/ivPDtOJhTwyNzHQe6c2-vQ/_open?pretty" -k
{
  "error" : {
    "root_cause" : [
      {
        "type" : "security_exception",
        "reason" : "missing authentication credentials for REST request [/ivPDtOJhTwyNzHQe6c2-vQ/_open?pretty]",
        "header" : {
          "WWW-Authenticate" : [
            "Basic realm=\"security\" charset=\"UTF-8\"",
            "Bearer realm=\"security\"",
            "ApiKey"
          ]
        }
      }
    ],
    "type" : "security_exception",
    "reason" : "missing authentication credentials for REST request [/ivPDtOJhTwyNzHQe6c2-vQ/_open?pretty]",
    "header" : {
      "WWW-Authenticate" : [
        "Basic realm=\"security\" charset=\"UTF-8\"",
        "Bearer realm=\"security\"",
        "ApiKey"
      ]
    }
  },
  "status" : 401
}

**Reply by dougburks** [reply]
Take a look at how `/usr/sbin/so-elasticsearch-query` authenticates to Elasticsearch.

**Reply by palitech100** [reply]
Thank you Doug! I was able to get the index open. Can you please confirm where I add the exception to make sure that index stays open. Would it be this file?  /opt/so/conf/curator/action/so-elasticsearch-close.yml and add the index name to "exclude:"

**Reply by dougburks** [reply]
You don't want to edit anything in `/opt/so/conf/curator/action/` directly as this configuration is managed by salt:
https://docs.securityonion.net/en/2.3/curator.html#configuration

**Reply by palitech100** [reply]
Thank you, Doug. I really appreciate all your help!! I was able to change the frequency of the curatorclose cron job managed by Salt and kept the indices open long enough to complete my traffic analysis. :)
